### ISSUE TITLE
Show the Intro Once Not working

### ISSUE THREAD

**Issue by kennyung6** [PaoloRotolo/AppIntro#245]
Please check off the following things BEFORE submitting a new issue. Any issues that do not properly follow this format will be closed without discussion. If you're requesting to have your app added to the list, you can erase the list and provide your app's name and play store listing link. Thanks!

[ ] What you've tried (include code snippets, or whole files as necessary):

added the Show intro once example code to my splash screen

```java
 //  Declare a new thread to do a preference check
        Thread t = new Thread(new Runnable() {
            @Override
            public void run() {
                //  Initialize SharedPreferences
                SharedPreferences getPrefs = PreferenceManager
                        .getDefaultSharedPreferences(getBaseContext());

                //  Create a new boolean and preference and set it to true
                boolean isFirstStart = getPrefs.getBoolean("firstStart", true);

                //  If the activity has never started before...
                if (isFirstStart) {

                    //  Launch app intro
                    Intent i = new Intent(SplashActivity.this, app_intro.class);
                    startActivity(i);

                    //  Make a new preferences editor
                    SharedPreferences.Editor e = getPrefs.edit();

                    //  Edit preference to make it false because we don't want this to run again
                    e.putBoolean("firstStart", false);

                    //  Apply changes
                    e.apply();
                }
            }
        });

        // Start the thread
        t.start();
```

[ ] What you thought should happen:

It should move from splashscreen activity to Main activity after the intro has been shown once

[ ] What actually happened:

unfortunately, it doesn't go beyond the splashscreen activity. it freezes.

**Comment by avluis** [comment]
Look at your intent, what is this?
`Intent(SplashActivity.this, app_intro.class);`

Can you show more of your code as well.

**Comment by kennyung6** [comment]
Okay Below is my Splash Activity and app_intro activity class.

Again, The Splash Activity is the first activity to be launched at start, then the app_intro activity (show only once - For 1st time use only), then to MainActivity. Thanks

# **SplashActivity.java**
`public class SplashActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Intent i = new Intent(SplashActivity.this, app_intro.class);
        startActivity(i);

    }
}`

# **app_intro.java**

`public class app_intro extends AppIntro {

    @Override
    protected void onCreate (Bundle savedInstance) {

        super.onCreate(savedInstance);

        //Default App intro Settings
        addSlide(SampleSlide.newInstance(R.layout.intro));
        addSlide(SampleSlide.newInstance(R.layout.intro2));
        addSlide(SampleSlide.newInstance(R.layout.intro3));
        addSlide(SampleSlide.newInstance(R.layout.intro4));

        //Set Slide Animation
        setFadeAnimation();

        //Turn Vibration on & Intensity
        setVibrate(true);
        setVibrateIntensity(30);

        //setBarColor(Color.parseColor("#3F51B5"));
        setSeparatorColor(Color.parseColor("#2196F3"));
    }

    private void loadMainActivity() {

        Intent intent = new Intent(this, MainActivity.class);
        startActivity(intent);
    }

    @Override
    public void onSkipPressed(Fragment currentFragment) {
        super.onSkipPressed(currentFragment);
        loadMainActivity();
        Toast.makeText(getApplicationContext(),getString(R.string.skip), Toast.LENGTH_SHORT).show();
    }

    @Override
    public void onDonePressed(Fragment currentFragment) {
        super.onDonePressed(currentFragment);
        loadMainActivity();
    }
}
`

**Comment by avluis** [comment]
Haven't looked at your code yet but I edited your posts for better legibility - check them out for your next comment.

**Comment by kennyung6** [comment]
Okay... so whats next?

**Comment by avluis** [comment]
Mind if I ask, where does the code in the opening comment come along?

**Comment by kennyung6** [comment]
If you mean the show intro_once code, i saw it on the Wiki Page for the app_intro Library

**Comment by avluis** [comment]
No no no - in the context of this issue, where does the code in the first comment go in the code shown afterwards.
I asked for you to show code, not snippets.

**Comment by kennyung6** [comment]
Oh okay.. From the splash activity it then goes to the MainActivity

**Comment by avluis** [comment]
Referencing this issue as it is related #250
If this issue has not being fixed, please reopen with something substantial.
